### PR TITLE
ENT-9028: Skip two tests on older platforms due to lack of fix there (3.18)

### PR DESCRIPTION
**Pull request by craigcomstock**
ENT-5866 fix for set_line_based() does not work when combined
with edit_defaults => empty was only back ported to 3.18.x.

Ticket: ENT-9028
Changelog: none
(cherry picked from commit 636908732ff086d48cd59b861641b2d294ef6210)

### CODE REVIEW

**Comment by nickanderson** [comment]
Waiting for https://github.com/cfengine/masterfiles/pull/2441 to be merged, then will adapt this PR.

**Comment by nickanderson** [comment]
@cf-bottom jenkins, please

**Comment by craigcomstock** [comment]
https://github.com/cfengine/masterfiles/pull/2440 is waiting on this one somewhat.

**Comment by nickanderson** [comment]
@cf-bottom jenkins, please

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/8384/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/8384/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-8384/

**Review by nickanderson — APPROVED** [review]
LGTM, but hey most recent commits were mine

File: .travis.yml [hunk 1]
```diff
@@ -19,17 +19,28 @@ matrix:
       script:
         - sudo ./travis-scripts/valgrind.sh
 
+    # JOB
```
Comment (craigcomstock) [inline .travis.yml:22]: just delete it? maybe leave a comment about why NOT to test against master binaries?

Comment (nickanderson) [reply .travis.yml:22]: k

**Comment by nickanderson** [comment]
@cf-bottom jenkins, please

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/8385/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/8385/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-8385/

File: lib/files.cf [hunk 2]
```diff
@@ -778,7 +778,15 @@ bundle edit_line set_line_based(v, sep, bp, kp, cp)
       unless => strcmp( "true", $(edit.empty_before_use) );
 @endif
 
-    (cfengine_3_15|cfengine_3_16|cfengine_3_17|cfengine_3_18|cfengine_3_19|cfengine_3_20)::
+@if minimum_version(3.18)
```
Comment (nickanderson) [inline lib/files.cf:781]: @craigcomstock note this change to `set_line_based`, Ill need to pick it back to master.

Comment (nickanderson) [reply lib/files.cf:781]: Here https://github.com/cfengine/masterfiles/pull/2443

**Review by nickanderson — APPROVED** [review]

**Comment by craigcomstock** [comment]
Missed a later commit that was needed for completeness here: https://github.com/cfengine/masterfiles/pull/2444

File: lib/files.cf [hunk 3]
```diff
@@ -778,7 +778,15 @@ bundle edit_line set_line_based(v, sep, bp, kp, cp)
...
+    !(cfengien_3_18_0|cfengine_3_18_1|cfengine_3_18_2)::
```
Comment (craigcomstock) [inline lib/files.cf:782]: oops, mis-spelled, cfengien, will make a PR

Comment (craigcomstock) [reply lib/files.cf:782]: https://github.com/cfengine/masterfiles/pull/2445

**Comment by craigcomstock** [comment]
saw one typo so made a follow up PR: https://github.com/cfengine/masterfiles/pull/2445

**Pull request merged**